nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/EvaluateXPath.java

```diff
@@ -219,6 +239,25 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
         }
 
         final ComponentLog logger = getLogger();
+        final DocumentBuilderFactory docBuilderFactory=docBuilderFactoryRef.get();
+        final DocumentBuilder docBuilder;
+        try {
+            docBuilder = docBuilderFactory.newDocumentBuilder();
+
+            if (!context.getProperty(VALIDATE_DTD).asBoolean()) {
+                docBuilder.setEntityResolver(new EntityResolver() {
+                    @Override
+                    public InputSource resolveEntity(String publicId, String systemId) throws SAXException, IOException {
+                        return new InputSource(new StringReader(""));
+                    }
+                });
+            }
+        } catch (ParserConfigurationException e) {
+            logger.error("Configuration error while constructing DocumentBuilder {}", new Object[]{e});
+            session.transfer(flowFiles, REL_FAILURE);
+            return;
```

Comment:
@ijokarumawak Brilliant. That makes a lot of sense and I understand the spirit of it. Also, with the new change, there is a place where I am instantiating the XMLReader and that throws a SAX Exception.  I feel that there is no way to recover from it and have transferred the flow to FAILURE.  Let me know if that ought to be changed.